x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -103,18 +103,24 @@ protected List<Batch<LogicalPlan>> batches() {
         return rules();
     }
 
-    protected static Batch<LogicalPlan> substitutions() {
+    protected static Batch<LogicalPlan> transformStats() {
         return new Batch<>(
-            "Substitutions",
-            Limiter.ONCE,
-            new ReplaceLookupWithJoin(),
```

Comment:
@alex-spies Agreed, let's see if we can make it work without introducing a new batch running in a loop.

Reply:
Fyi, @dnhatn introduced a [new functionality to `metrics` source command](https://github.com/elastic/elasticsearch/pull/109979/files#diff-45f172b501f6b73039f2a486c29da73ed84d03468540659d86dfa8e10222ade9) and with that he also changed the LogicalPlanOptimizer by adding one new rule and calling again `ReplaceStatsNestedExpressionWithEval`.

Reply:
Okay, since we have another run of `ReplaceStatsNestedExpressionWithEval` now, I think this PR may not need to introduce the new batch anymore. (I'd still like to clean up the redundancy between surrogate substitution and `ReplaceStatsNestedExpressionWithEval`, but it's outside this PR's scope IMO.)